File: lab3/cs/DiffieHellman/dh/GaloisFieldFactory.cs
namespace dh;

using dh.Interfaces;

public class GaloisFieldFactory : IGaloisFieldFactory<GaloisField> {
    public GaloisField FromLong(long value) {
        return new GaloisField(value);
    }

    public GaloisField Clone(GaloisField instance) {
        return new GaloisField(instance);
    }
}

File: lab3/java/DiffieHellman/src/main/java/com/jpp_lab3/app/GaloisFieldFactory.java
package com.jpp_lab3.app;

import com.jpp_lab3.app.interfaces.IGaloisFieldFactory;

public class GaloisFieldFactory implements IGaloisFieldFactory<GaloisField> {
    @Override
    public GaloisField fromLong(final long value) {
        return new GaloisField(value);
    }

    @Override
    public GaloisField clone(final GaloisField instance) {
        return new GaloisField(instance);
    }
}
